experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -585,6 +591,17 @@ const mapStaticDispatchMap: PropertyDescriptorMap = {
 			return node.keys();
 		},
 	},
+	set: {
+		value(
+			this: SharedTreeMap<MapSchema>,
+			key: string,
+			content: ProxyNodeUnion<AllowedTypes, "javaScript">,
+		): SharedTreeMap<MapSchema> {
+			const node = getMapNode(this);
+			node.set(key, extractFactoryContent(content as any)); // TODO: why is this being treated as `unknown`?
```

Comment:
It's because `content` is typed over `AllowedTypes`, which is the super type of all allowed types. Normally you'd type it over a `TSchema extends MapSchema` (e.g. `ProxyField<TSchema["mapFields"]>`) and get the more specific type, but you don't have a `TSchema` here.

Reply:
```suggestion
			node.set(key, extractFactoryContent(content as any));
```